### DISCUSSION TITLE
Combining ROI activity from a set of ROIs

### DISCUSSION THREAD

**Discussion by jbscherrer** [bonsai-rx/bonsai#1370] [Q&A]
I have an RoiActivity node that is finding the ROI activity of several different ROIs simultaneously and outputs a RegionActivityCollection object. I would like to create an array corresponding to each IplImage input that contains the activity of each ROI, so I can then sum or average them together and also save them to all to a csv. Is there a straightforward way to do this? I can use a ConvertToArray to convert the RegionActivityCollection to an array of RegionActivity, but can't figure out how to create an array containing just the Activity.Val0 property for each RegionActivity.

**Comment by bruno-f-cruz** [comment]
I am not sure I completely follow what you are looking for. Can you provide a workflow with an example of something close to what you want to achieve? 
Thanks!

**Reply by jbscherrer** [reply]
Here's a screenshot of the workflow linked below--the hacky solution depicted only incorporates the first three ROIs from each `RoiActivityCollection`

![image](https://github.com/bonsai-rx/bonsai/assets/39060975/193881bd-5bc9-4934-bb7b-6914e642f21c)

**Comment by jbscherrer** [comment]
Here's an example workflow that does what I want with only the first three ROIs for each of two sets of ROIs read in from .csv files. Right now I'm manually selecting elements 1, 2, and 3 from the RoiActivityCollection using individual Index nodes.

[Ensemble_Activity.zip](https://github.com/bonsai-rx/bonsai/files/11536536/Ensemble_Activity.zip)

**Comment by jbscherrer** [comment]
Figured it out with the help of Andrew Erskine. For anyone trying to accomplish the same task:

![image](https://github.com/bonsai-rx/bonsai/assets/39060975/e7cb4e36-0207-4416-85a2-ff5ed67acb76)

And within `roi_to_array`:
![image](https://github.com/bonsai-rx/bonsai/assets/39060975/13df28bc-8d34-4693-b252-bd759041c093)

My understanding is that this works because Merge can also operate on `Enumerable` objects (which `RegionActivityCollection` implements according to the docs) in the same way that it operates on `Observable` sequences.
